contracts/interfaces/IUniAgent.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+/// @title IUniAgent Interface
+/// @author imToken Labs
+interface IUniAgent {
+    error ZeroAddress();
+    error InvalidMsgValue();
+    error UnknownRouterType();
+
+    event Swap(address indexed user, address indexed router, address indexed inputToken, uint256 inputAmount);
+
+    enum RouterType {
+        v2,
+        v3,
+        universal
```

Comment:
nit: I personally would prefer using pascal case for enum key.

Reply:
Ok, should be easier to identify. Adjusted in f1d1078.